File: src/main/java/com/dentistPlus/dentist_backend/model/Patient.java
package com.dentistPlus.dentist_backend.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;

@Data
@Entity
@Table(name = "Patient")
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String avatar;
    private String name;
    private String surname;
    private String pesel;
    private LocalDate dateOfBirth;
    private String phoneNumber;
    private String email;

    public Patient() {
    }

    public void setValues(Patient newPatient)
    {
        this.setName(newPatient.getName());
        this.setSurname(newPatient.getSurname());
        this.setPesel(newPatient.getPesel());
        this.setPhoneNumber(newPatient.getPhoneNumber());
        this.setDateOfBirth(newPatient.getDateOfBirth());
    }
}

File: src/model/patient/Patient.ts
import mongoose from 'mongoose';
import { Tooth, toothSchema } from './Tooth';

export interface Patient {
  avatar: string;
  name: string;
  surname: string;
  pesel: string;
  dateOfBirth: Date;
  email: string;
  phoneNumber: string;
  teeth: Tooth[];
}

export interface PatientModel extends Patient, Document {};

const patientSchema = new mongoose.Schema({
  name: { type: String, required: true },
  surname: { type: String, required: true },
  avatar: { type: String, required: false },
  pesel: {  type: String, required: true, unique: true },
  dateOfBirth: { type: Date, required: true },
  email: { type: String, required: true },
  phoneNumber: { type: String, required: true },
  teeth: {type: [toothSchema], required: false}
})

export default mongoose.model<PatientModel>('Patient', patientSchema);